tests/ui/conftest.py

```diff
@@ -1,37 +1,85 @@
 import datetime
 import json
 import os
+import random
 import urlparse
 
 import jwt
 import pytest
 import requests
 from django.core.management import call_command
 from fxapom.fxapom import DEV_URL, PROD_URL, FxATestAccount
-from olympia.amo.tests import create_switch
+from olympia import amo
+from olympia.addons.forms import icons
+from olympia.addons.models import AddonUser, Preview
+from olympia.addons.utils import generate_addon_guid
+from olympia.amo.tests import (
+    addon_factory,
+    create_switch,
+    user_factory,
+    version_factory,
+)
+from olympia.constants.applications import APPS, FIREFOX
+from olympia.constants.base import (
+    ADDON_EXTENSION,
+    ADDON_PERSONA,
+    STATUS_PUBLIC
+)
+from olympia.landfill.collection import generate_collection
+from olympia.landfill.generators import generate_themes
+from olympia.reviews.models import Review
+from olympia.users.models import UserProfile
 from pytest_django import live_server_helper
 
 
 @pytest.fixture(scope='function')
-def my_base_url(base_url, request):
-    return base_url or request.getfixturevalue("live_server").url
+def my_base_url(base_url, request, pytestconfig):
+    """
+    Base URL used to start the 'live_server'.
+    """
+
+    if base_url or 'PYTEST_BASE_URL' in os.environ:
+        pytestconfig.option.usingliveserver = False
+        return base_url or os.environ.get('PYTEST_BASE_URL')
+    else:
+        pytestconfig.option.usingliveserver = True
+        return request.getfixturevalue("live_server").url
+
+
+def pytest_cmdline_preparse(args):
+    """
+    Fixture for running the tests in serial or parallel depending on the
+    base URL.
+    """
+    if 'PYTEST_BASE_URL' in os.environ:
+        return
+    else:
+        args[:] = ["--boxed"] + args
```

Comment:
I don't feel like we should do this. We can leave it to the user to specify if they want tests to be boxed or run in parallel. For Travis, we should explicitly add the --boxed argument.